test/test_torch.py

```diff
@@ -3874,10 +3874,21 @@ def test_diagflat(self):
         self._test_diagflat(self, dtype=torch.float32, device='cpu')
 
     def test_eye(self):
-        res1 = torch.eye(100, 100)
-        res2 = torch.Tensor()
-        torch.eye(100, 100, out=res2)
-        self.assertEqual(res1, res2)
+        for dtype, device in product(torch.testing.get_all_dtypes(), torch.testing.get_all_device_types()):
```

Comment:
This doesn't seem having `half` tested

Reply:
get_all_dtypes() contains returns a list that contains torch.float16.

https://github.com/pytorch/pytorch/blob/465b4de9d4378ac58013b56ea45fe30f9beadb39/torch/testing/__init__.py#L89-L91

Reply:
Yes, you are right.